docs/sapling.md

```diff
@@ -0,0 +1,277 @@
+The `prepareShieldedTransaction` method returns the crafted sapling transaction parameter and does not perform any change on the shielded pool. A subsequent step where the sapling transaction parameter is submitted to the smart contract must be done. Note that in a case of a shielded transaction, the shielded amount must be sent along when calling the smart contract to transfer the tez to the shielded pool, or it will result in an error.
+
+Here is an example of how to prepare and inject a shielded transaction using Taquito:
+
+```ts
+import { TezosToolkit } from '@taquito/taquito';
+import { SaplingToolkit } from '@taquito/sapling';
+
+const tezos = new TezosToolkit('https://jakartanet.ecadinfra.com/');
+// set up your signer on the TezosToolkit as usual
+const saplingContract = await tezos.contract.at('KT1UYwMR6Q6LZnwQEi77DSBrAjKT1tEJb245');
+
+const inMemorySpendingKey = await InMemorySpendingKey.fromMnemonic('YOUR_MNEMONIC');
+
+const saplingToolkit = new SaplingToolkit(
+    inMemorySpendingKey, 
+    { contractAddress: saplingContract.address, memoSize: 8 }, 
+    tezos.getFactory(RpcReadAdapter)()
+)
+
+// Fetch a payment address (zet)
+const inMemoryViewingKey = await inMemorySpendingKey.getInMemoryViewingKey();
+const paymentAddress = (await inMemoryViewingKey.getAddress()).address;
+
+// prepare the shielded transaction
+const shieldedTx = await saplingToolkit.prepareShieldedTransaction([{
+    to: paymentAddress,
+    amount: 3,
+    memo: 'test',
+    mutez: false // set to false by default
+}])
+
+// Inject the sapling transaction using the ContractAbstraction
+// The amount MUST be specified in the send method to transfer the 3 tez to the shielded pool
+const op = await saplingContract.methods.default([shieldedTx]).send({ amount: 3 });
+await op.confirmation();
+```
+
+## How to prepare a sapling transaction?
+
+A sapling transaction allows sending tokens from an address (zet) to an address (zet). The `prepareSaplingTransaction` method of the `SaplingToolkit` takes an array of `ParametersSaplingTransaction`, making it possible to send tez to multiple addresses at once if needed. 
+
+The `ParametersSaplingTransaction` is an object made of:
+- a `to` property, which is the destination address (zet)
+- an `amount` property, which is the amount to shield in tez by default
+- an optional `memo` that cannot be longer than the specified memo size
+- an optional `mutez` property that must be set to true if the specified amount is in mutez rather than tez
+
+The `prepareSaplingTransaction` method returns the crafted sapling transaction parameter and does not perform any change on the shielded pool. A subsequent step where the sapling transaction parameter is submitted to the smart contract must be done. 
+
+:::note
+A user should not use its own implicit account (tz1, tz2, tz3) to submit a sapling transaction but rather have a third party inject it.
+:::
+
+Here is an example of how to prepare and inject a sapling transaction using Taquito:
```

Comment:
for consistency should all sapling be Sapling?

Reply:
Ok I renamed all sapling to Sapling